### DISCUSSION TITLE
ComputeError: reading computation output: Array { inner_type: SecretInteger, size: 1 } is not a primitive type.

### DISCUSSION THREAD

**Discussion by Devika-Pardeshi** [NillionNetwork/discussions#47] [General]
Hi,                            
I am trying to add three secrets to a variable and store it in an array. But facing following error in client.

**Nada Program -**

    from nada_dsl import *
    
    def nada_main():

    party1 = Party(name="Party1")

    my_int1 = SecretInteger(Input(name="my_int1", party=party1))

    my_int2 = SecretInteger(Input(name="my_int2", party=party1)) 

    my_int3 = SecretInteger(Input(name="my_int3", party=party1))    

    addition = my_int1 + my_int2 + my_int3
   
    my_array = Array.new(addition)

    return [Output(my_array, "my_output", party1)]
  

**Client Program -**

      from pdb import set_trace as bp
      import asyncio
      import json
      import os
      import py_nillion_client
      
      SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
      with open(os.path.join(SCRIPT_DIR, "local.json"), "r") as fp:
          config = json.load(fp)      
      
      async def main():
          bootnodes = config["bootnodes"]
          cluster_id = config["cluster_id"]
  
      payments_config = py_nillion_client.PaymentsConfig(
          config["payments_config"]["rpc_endpoint"],
          config["payments_config"]["signer"]["wallet"]["private_key"],
          int(config["payments_config"]["signer"]["wallet"]["chain_id"]),
          config["payments_config"]["smart_contract_addresses"]["payments"],
          config["payments_config"]["smart_contract_addresses"][
              "blinding_factors_manager"
          ],
      )
  
      nodekey = py_nillion_client.NodeKey.from_file(config["keypath"]["node"])
      userkey = py_nillion_client.UserKey.from_file(config["keypath"]["wuser"])
  
      # Create Nillion Client
      client = py_nillion_client.NillionClient(
          nodekey,
          bootnodes,
          py_nillion_client.ConnectionMode.relay(),
          userkey,
          payments_config,
      )
  
     
      my_int1 = py_nillion_client.SecretInteger(42)
     
      my_int2 = py_nillion_client.SecretInteger(24)
  
      my_int3 = py_nillion_client.SecretInteger(24)
   
      # We bind the storage of the secret to the circuit and the concrete party
      my_party_id = (
          client.party_id() if callable(getattr(client, "party_id")) else client.party_id
      )
  
      program_id = config["programs"]["basic"]
      bindings = py_nillion_client.ProgramBindings(program_id) 
     
      combine_secrete = py_nillion_client.Secrets({"my_int1": my_int1 ,"my_int2": my_int2 , "my_int3": my_int3})
  
       # Store the secret
      store_id = await client.store_secrets(
          cluster_id, bindings, combine_secrete, None
      )
      print(f"Stored secret, store_id ={store_id}")
  
      # bind the parties in the computation to the client
      bindings = py_nillion_client.ProgramBindings(program_id)
      bindings.add_input_party("Party1", my_party_id)
      bindings.add_output_party("Party1", my_party_id)
  
      print(f"Computing using program {program_id}")
      print(f"Stored secret: {store_id}")
      print(f"Provided secret: {combine_secrete}")
      # do the computation
      compute_id = await client.compute(
          cluster_id,
          bindings,        
          [store_id],
          combine_secrete,        
          py_nillion_client.PublicVariables({}),
      )
  
      print(f"Computation sent to the network, compute_id = {compute_id}")
      print("Waiting computation response")
      while True:
          event = await client.next_compute_event() 
          
              
          if isinstance(event, py_nillion_client.ComputeFinishedEvent):
              result = event.result.value
              print(
                  f"Received computation result for {event.uuid}, value = {event.result.value}"
              )
              break
  
      print(f"Output of cricuit 1 is , {result}") 
  
  asyncio.run(main())

**Error -**
  event = await client.next_compute_event() 
 |py_nillion_client.ComputeError: reading computation output: Array { inner_type: SecretInteger, size: 1 } is not a primitive type

Tried with  casting nada program array initialization to integer as below
my_array = Array.new(Integer(addition))

then we are getting below compilation error,
nillion  |         raise ValueError(f"Expected int, got {type(value)._name_}")
nillion  |     ValueError: Expected int, got SecretInteger

**Question :** 
           Need input on handling primitive types at client side?

**Comment by oceans404** [comment]
The issue is with Arrays here in your Nada program

```
my_array = Array.new(addition)
```

Instead you should use a Python list. Check out the examples here that go into the use of Python lists that contain secret integer values: https://docs.nillion.com/nada-lang-tutorial-lists-and-comprehensions

**Comment by Davetbutler** [comment]
Hey @Devika-Pardeshi.

Dave from the product team here. 

I was wondering how your journey building with Nillion is going? Do you need help with anything, or have any feedback for us?
